framework/src/convergence/ResidualConvergence.C

```diff
@@ -0,0 +1,360 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "ResidualConvergence.h"
+#include "FEProblemBase.h"
+#include "PetscSupport.h"
+#include "Executioner.h"
+#include "PerfGraphInterface.h"
+#include "NonlinearSystemBase.h"
+#include "ActionWarehouse.h"
+
+// PETSc includes
+#include <petscsnes.h>
+#include <petscksp.h>
+#include <petscdm.h>
+
+// PetscDMMoose include
+#include "PetscDMMoose.h"
+
+registerMooseObject("MooseApp", ResidualConvergence);
+
+InputParameters
+ResidualConvergence::validParams()
+{
+  InputParameters params = Convergence::validParams();
+  params += FEProblemSolve::commonParams();
+
+  params.addClassDescription(
+      "Check the convergence of residual during nonlinear iterations. Return "
+      "CONVERGED if the residual is less than user-supplied stopping criteria,"
+      "DIVERGED if the residual is greater than the tolerance, or ITERATING otherwise.");
```

Comment:
```suggestion
      "Checks convergence based on absolute and relative error of the residual.");
```

Reply:
What you've described is true of all Convergence objects, not this one in particular